### DISCUSSION TITLE
How to override `stdenv`?

### DISCUSSION THREAD

**Discussion by rushmorem** [ipetkov/crane#99] [Q&A] [answered]
I tried

- passing it in both `buildPackage` and `buildDepsOnly`, ie `craneLib.buildPackage { stdenv = pkgs.gcc10Stdenv; }`, and
- overriding the derivations in both, ie `(buildPackage { ... }).overrideAttrs (old: { stdenv = pkgs.gcc10Stdenv; })`

In both cases, it's getting ignored.

Edit: I just tried `overrideScope'`, it's being ignored too:-

```nix
craneLib = (crane.mkLib pkgs).overrideScope' (final: prev: {
    cargo = rustToolchain;
    rustc = rustToolchain;
    stdenv = pkgs.gcc10Stdenv;
});
```

**Comment by ipetkov** [comment, marked as answer]
Hmm that's really weird that `overrideScope'` isn't working either...

I don't mind allowing `stdenv` overrides more easily in `mkCargoDerivation`, will put a PR up shortly

**Reply by rushmorem** [reply]
Thank you!

**Reply by ipetkov** [reply]
Here's the PR if you'd like to give it a spin: https://github.com/ipetkov/crane/pull/101

**Reply by rushmorem** [reply]
Sure! Thanks.

**Reply by rushmorem** [reply]
~~@ipetkov It's still being ignored on `github:ipetkov/crane/override-stdenv`.~~

I have `stdenv = gcc10Stdenv;` in my `buildPackage` but it's still trying to build with GCC 11.3.

```
$ gcc --version
gcc (GCC) 11.3.0
```

Edit: Never mind. It's working fine for me now.

**Comment by dpc** [comment]
@rushmorem Can you please let me know what are you doing? I'm just curious / trying to learn more about Nix+Rust interactions.

**Reply by rushmorem** [reply]
@dpc I'm packaging a crate that depends on `grpcio-sys`. That crate contains C source code that's not compatible with `gcc 11.3`, hence the need to use `gcc 10`. I could bypass the source code, and the need for an older `gcc` by setting `GRPCIO_SYS_USE_PKG_CONFIG = "1"` and adding `grpc` to my flake dependencies but that's giving me linker issues at the end.
